bash_automation.md

```diff
@@ -0,0 +1,165 @@
+# Automating workflows using bash
+
+## Objectives
+
++ Write a shell script with multiple variables.
++ Incorporate a for loop into a shell script.
++ Understand how to make a workflow more efficient and less error-prone
+
+
+We have now worked through two workflows, RNA-seq analysis and variant calling.
+In both cases, we pasted all of our commands into the terminal, and used for 
+loops to run each step of the workflow over our six samples. This is a great way
+to understand how each tool works and to establish a workflow you want to use.
+
+Let's imagine now that we decided to download all 672 samples from this dataset
+instead of only working with our six samples. We already know how each tool in
+our workflow works, and we know how to run them. We can now use bash scripting
+to automate our workflow. 
+
+We are going to automate the quality control portion of our workflow. To write 
+a script to run our FastQC analysis, we’ll take each of the commands we entered 
+to run FastQC and process the output files and put them into a single file with 
+a `.sh` extension. The `.sh` is not essential, but serves as a reminder to 
+ourselves and to the computer that this is a shell script.
+
+## Starting with automation
+
+We will be constructing our bash script in RStudio. We will construct our bash
+script in a text file in the "script" pane of RStudio, and we will execute it 
+in the "Terminal" tab of the "console" pane. We could use nano, but RStudio
+makes it easier to navigate a text file. 
+
+First, let's make a new directory called `scripts`.
+
+```
+cd ~
+mkdir scripts
+```
+
+## Constructing a bash script
+
+When we write a bash script, we need to add *all* commands that we ran in our
+workflow. This includes making and changing directories, moving files, and 
+running analysis programs like `fastqc`. 
+
+Let's start by changing into the right directory and running `fastqc`:
+
+```
+cd ~/data/
+
+fastqc *fastq.gz
+```
+
+That's it! We now have a bash script that automates `fastqc`! 
+
+Let's run the bash script from our terminal in RStudio
```

Comment:
Why RStudio terminal instead of regular terminal? They've been ssh-ing in each day, maybe better to keep RStudio for R and command line at the terminal?

Reply:
This was a suggestion by @sateeshperi for snakemake, so I tried to do the same thing as he suggested there. I don't have a strong preference.

Reply:
the idea with Rstudio, was to have a terminal and text editor in one window hoping it would make it easier to follow. no preference actually.